react/features/blur/components/VideoBlurButton.js

```diff
@@ -0,0 +1,89 @@
+// @flow
+
+import { translate } from '../../base/i18n';
+import { connect } from '../../base/redux';
+import { AbstractButton } from '../../base/toolbox';
+import type { AbstractButtonProps } from '../../base/toolbox';
+import { toggleBlurEffect } from '../actions';
+import { isVideoBlurEnabled } from '../functions';
+
+/**
+ * The type of the React {@code Component} props of {@link VideoBlurButton}.
+ */
+type Props = AbstractButtonProps & {
+
+    /**
+     * True if the video background is blurred or false if it is not.
+     */
+    _isVideoBlurred: boolean,
+
+    /**
+     * The redux {@code dispatch} function.
+     */
+    dispatch: Function
+};
+
+/**
+ * An abstract implementation of a button for accessing settings.
+ */
+class VideoBlurButton extends AbstractButton<Props, *> {
+    accessibilityLabel = 'toolbar.accessibilityLabel.videoblur';
+    iconName = 'icon-settings';
+    label = 'toolbar.startvideoblur';
+    tooltip = 'toolbar.startvideoblur';
+    toggledLabel = 'toolbar.stopvideoblur';
+
+    /**
+     * Handles clicking / pressing the button, and opens the appropriate dialog.
+     *
+     * @protected
+     * @returns {void}
+     */
+    _handleClick() {
+        const {
+            _isVideoBlurred,
+            dispatch } = this.props;
+
+        if (_isVideoBlurred) {
+            dispatch(toggleBlurEffect(!_isVideoBlurred));
+        } else {
+            dispatch(toggleBlurEffect(!_isVideoBlurred));
+        }
+    }
+
+    /**
+     * Helper function to be implemented by subclasses, which must return a
```

Comment:
Update the docs according to the specific use case here.

Reply:
This comment is very generic and it is copied from AbstractButton class. Please add a description that explains what `_isToggled` is implemented to do in `VideoBlurButton` class! Maybe 'Returns the video blur status' or something like this...

Reply:
Updated comment